homeassistant/components/rova/__init__.py

```diff
@@ -1 +1,55 @@
 """The rova component."""
+
+from __future__ import annotations
+
+from requests.exceptions import ConnectTimeout, HTTPError
+from rova.rova import Rova
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import (
+    CONF_HOUSE_NUMBER,
+    CONF_HOUSE_NUMBER_SUFFIX,
+    CONF_ZIP_CODE,
+    DOMAIN,
+    LOGGER,
+)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up ROVA from a config entry."""
+
+    api = Rova(
+        entry.data[CONF_ZIP_CODE],
+        entry.data[CONF_HOUSE_NUMBER],
+        entry.data[CONF_HOUSE_NUMBER_SUFFIX],
+    )
+
+    try:
+        rova_area = await hass.async_add_executor_job(api.is_rova_area)
+    except (ConnectTimeout, HTTPError) as ex:
+        LOGGER.error("Could not retrieve details from ROVA API")
+        raise ConfigEntryNotReady from ex
+
+    if not rova_area:
+        LOGGER.error("Rova does not collect garbage in this area")
+        return False
```

Comment:
raise a ConfigEntryError instead.

In the future we should create an issue of some sort for the user to act on. (I can assume that Rova will gain new areas and lose some areas in the future, causing existing setups to fail)